Source/Server/Startup.cs

```diff
@@ -45,20 +56,39 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 app.UseDeveloperExceptionPage();
             }
             app.UseSwagger();
-            app.UseSwaggerUI(c =>
-            {
-                c.SwaggerEndpoint("/swagger/v1/swagger.json", "Runtime API v1");
-            });
+            app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "Runtime API v1"));
 
+            app.UseDefaultFiles();
             app.UseStaticFiles();
 
             app.UseGrpcWeb();
             app.UseCors();
             app.UseRouting();
+
+            app.UseWebSockets();
+
             app.UseEndpoints(endpoints =>
             {
+                const string graphQLRoute = "/graphql";
+                const string graphQLPlaygroundRoute = "/graphql/ui";
+
                 endpoints.MapControllers();
+                endpoints.MapGraphQLPlayground(
+                    new PlaygroundOptions
+                    {
+                        GraphQLEndPoint = graphQLRoute,
+                        SubscriptionsEndPoint = graphQLRoute
+                    }, graphQLPlaygroundRoute);
+                endpoints.MapGraphQL(graphQLRoute).WithOptions(new GraphQLServerOptions
+                {
+                    Tool = { Enable = false }
+                });
             });
+
+            app.UseProxyServer("/metrics", "http://localhost:9700/metrics");
```

Comment:
What is this - I don't really understand? Seems like you want to avoid CORS problems by proxying to itself onto the metrics port? It seems a bit vulnerable to hardcode the port like this here.

Can't we just serve the management UI and the metrics on the same port? That would make sense to me...

Or just let the management UI talk to the metrics directly on that port.

Reply:
They have now in 4.1.1 a new AspNet support - have fixed the code by upgrading to this and taking out this. Before they didn’t have this and it spun up its own Web server on the port you dedicated, so the proxy I put in was just to be able to go through the origin without thinking about CORS.